cdap-client-tests/src/test/java/co/cask/cdap/client/PreferencesClientTestRun.java

```diff
@@ -238,7 +239,11 @@ public void testPreferences() throws Exception {
       Assert.assertEquals(propMap, client.getApplicationPreferences(FAKE_APP_ID, false));
       Assert.assertEquals(propMap, client.getProgramPreferences(flow, false));
     } finally {
-      appClient.delete(FAKE_APP_ID);
+      try {
+        appClient.delete(FAKE_APP_ID);
+      } catch (ApplicationNotFoundException e) {
+        // ok if this happens, means its already deleted.
```

Comment:
its in a finally, so there could be an exception in the test that happens before its redeployed. This test will fail, but we'd want the failure exception to be from that earlier failure and not from this delete.